src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -102,6 +102,21 @@ def exit_code(self):
     return self._exit_code
 
 
+class _PantsProductPrecomputeFailed(Exception):
+  """
+  Represents a failure in the precompute step of v2 products for pants runs.
+  (e.g. The user called ./pants dependees and there was a broken link)
+
+  Will be raised by the DaemonPantsRunner at creation time.
+  """
+
+  def __init__(self, wrapped_exception):
```

Comment:
Would be wonderful to add type hints if you're willing to. Even if we can't add the tag `type_checked` yet due to pre-existing errors, it will help us in the future. You'd do this by finding what `wrapped_execution` is (an `Exception`?) and then this:

```python
def __init__(self, wrapped_exception: Exception) -> None:
  ...
```

Reply:
Yay!